app/controllers/concerns/api/participant_actions.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require "abstract_interface"
+
+module Api
+  module ParticipantActions
+    include AbstractInterface
+
+    def withdraw
+      perform_action(service_namespace: ::Participants::Withdraw)
+    end
+
+  private
+
+    def perform_action(service_namespace:)
+      render json: serialized_response(profile_for(service_namespace))
+    end
+
+    def recorder(service_namespace:)
+      "#{service_namespace}::#{::Factories::CourseIdentifier.call(course_identifier)}".constantize
+    end
+
+    def serialized_response(profile)
+      ParticipantSerializer
+        .new(profile.user)
+        .serializable_hash.to_json
+    end
```

Comment:
This looks ECF specific (cause `ParticipantSerializer` is outputting ECF shape), does it belong in this module?

Reply:
I could move this to another module and include it in `ParticipantController` and `ECFParticipantController`